trieste/models/interfaces.py

```diff
@@ -92,6 +94,27 @@ def get_observation_noise(self) -> TensorType:
         """
         raise NotImplementedError(f"Model {self!r} does not provide scalar observation noise")
 
+    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
```

Comment:
what about having a generic `sampler` method and you have to pass say a `ModelSampler` object (a parent abstract class for all samplers)?

depending on the need a user could then pass either an RFF based sampler or reparametrization based method or something else, if you don't use acq functions that need these then you don't pass anything

defining a method for each type of a sampler does not seem like a great solution, is there a problem with decoupling model and sampler implementation like this?

Reply:
Second that, method per sampler type doesn't sound great at the first glance